the-collab-lab/tcl-36-smart-shopping-list
# Issue #5 - Join an existing shopping list with a token

## Description

- Added new input for user to enter an existing token to join list.
- Check if token exists in Firestore - if the token exists, the user is redirected to List View. If not, the user has the option to try again or create a new list.
**Extras:** 
- User not able to switch to listView or addItemView if no token saved in localStorage

## Related Issue

closes #5 

## Acceptance Criteria

* When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<img width="306" alt="Screen Shot 2022-01-25 at 12 49 58 PM" src="https://user-images.githubusercontent.com/78327387/151057466-eaae63bc-5545-4b92-a65f-ff411b14ed21.png">



### After

<img width="561" alt="Screen Shot 2022-01-25 at 12 50 11 PM" src="https://user-images.githubusercontent.com/78327387/151057493-bb4a6174-2dc1-4ba3-a156-1c9b967e310a.png">

<img width="683" alt="after" src="https://user-images.githubusercontent.com/83471057/151633139-24dc0a21-0243-4f41-9bff-b14f702da20a.png">

## Testing Steps / QA Criteria

- After pulling down this branch to your local environment, run "npm i" to install dependencies, run ```npm start``` to run the application.
- Make sure you have cleared local storage by running ```localStorage.clear()``` in your browser console.
- Join an existing list by entering ```tin verb job``` in the input field. You can also check out other lists by going to our [Firestore](https://console.firebase.google.com/u/0/project/tcl-36-smart-shopping-list/firestore/data/~2Ftin%20verb%20job~2F90SXuoAv8ne0Gnfb1Uhr) console and entering one of the three word tokens. Make sure you are redirected to the list view!
- To see the error message, enter a token that does not exist (any random word) and you should see the notification appear.
- You can also create a new list and verify that it exists in your local storage and in Firestore under the 'token' collection.
- Adding items will create a new collection under the token name.
